src/main/java/org/hiero/sketch/table/LazyMembership.java

```diff
@@ -127,17 +108,104 @@ public IMembershipSet sample(final int k) {
      * Samples the base map for k items and then applies the filter on that set.
      * Makes samplingAttempts attempts to reach k samples
      * this way and then gives up and returns whatever was sampled.
-      */
+     * @param k the number of samples without replacement taken
+     * @param seed the seed for the random generator
+     * @return
+     */
     @Override
     public IMembershipSet sample(final int k, final long seed) {
         return sample(k, seed, true);
     }
 
+    /**
+     *
+     * @return An approximation of the size based on a smaple of size 20. May return
+     */
+    public int getApproxSize() {
+        if (this.rowCountCorrect)
+            return this.rowCount;
+        final IMembershipSet sampleSet = this.sample(20);
+        if (sampleSet.getSize() == 0)
+            return 0;
+        int snumber = 0;
+        final IRowIterator it = sampleSet.getIterator();
+        int curr = it.getNextRow();
+        while (curr >= 0) {
+            if (this.filter.test(curr))
+                snumber++;
+            curr = it.getNextRow();
+        }
+        if (this.baseMap instanceof LazyMembership)
```

Comment:
Are you assuming here that only LazyMembership has an expensive exactSize()?
If so you should probably document this at the very least.